File: 소스 코드/C#/PPT Remote Viewer Server/Utils/Connections/PacketFactory.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using PPTRemoteViewerServer.Utils.Statics;

namespace PPTRemoteViewerServer.Utils.Connections
{
    public class PacketFactory
    {
        public static byte[] CreateScreenPacket(Bitmap screen)
        {
            List<byte> packet = new List<byte>();
            packet.Add((byte)PacketType.Screen);

            byte[] jpegEncoded = BitmapManager.EncodeToJPEG(screen, 0.5, 0.5, 0.25);
            packet.AddRange(BitConverter.GetBytes(jpegEncoded.Length));
            packet.AddRange(jpegEncoded);

            return packet.ToArray();
        }
    }
}

File: Source code/Java test client/Java test client/src/pptRemoteViewerClient/utils/connections/PacketFactory.java
package pptRemoteViewerClient.utils.connections;

public class PacketFactory {
	public static final byte[] createSetPreviousPagePacket() {
		byte packetType = (byte)PacketType.Key.ordinal();		
		return new byte[] { packetType, 0 };
	}
	
	public static final byte[] createSetNextPagePacket() {
		byte packetType = (byte)PacketType.Key.ordinal();		
		return new byte[] { packetType, 1 };
	}
}
